python/hyperon/exts/agents/agent_base.py

```diff
@@ -40,7 +40,40 @@ def __next__(self):
             raise StopIteration
         return self._result.get()
 
-class AgentObject:
+
+class BaseListeningAgent:
```

Comment:
It doesn't look like `BaseListeningAgent` will be used by itself. Why not create a `ListeningAgent` which inherits from `AgentObject` instead of inheriting `AgentObject` from `BaseListeningAgent`? I'm not sure if the functionality from `BaseListeningAgent` is always needed and will not disturb some uses of `AgentObject`. Inheriting `ListeningAgent` from `AgentObject` looks more flexible to me. Or am I missing something?